linux_os/guide/system/network/network-firewalld/ruleset_modifications/ensure_firewall_rules_for_open_ports/oval/shared.xml

```diff
@@ -1,76 +1,183 @@
 <def-group>
-  <definition class="compliance" id="ensure_firewall_rules_for_open_ports" version="1">
-    {{{ oval_metadata("Make sure firewall rules exist for all open network ports, listening on non-loopback interfaces.") }}}
-    <criteria>
-      <criterion comment="Check firewall rules for udp ipv4 listening ports" test_ref="test_listening_fw_inet_ports_test" />
+  <definition class="compliance" id="{{{ rule_id }}}" version="1">
+    {{{ oval_metadata("Make sure firewall rules exist for all open network ports,listening on non-loopback interfaces.") }}}
+    <criteria operator="AND">
```

Comment:
You could also use the `{{% for proto in inet_protos %}}` jinja loop here to declare these tests.